vscode-lean4/src/leanclient.ts

```diff
@@ -216,18 +217,18 @@ export class LeanClient implements Disposable {
                     return;
                 },
 
-                didChange: (data, next) => {
-                    next(data);
+                didChange: async (data, next) => {
+                    await next(data);
                     if (!this.running || !this.client) return; // there was a problem starting lean server.
                     const params = this.client.code2ProtocolConverter.asChangeTextDocumentParams(data);
                     this.didChangeEmitter.fire(params);
                 },
 
-                didClose: (doc, next) => {
+                didClose: async (doc, next) => {
```

Comment:
this is probably due to the vscode version change, please revert this since Gabriel wants this to be done in a separate PR.